src/7-to-8/major-changes/compatibility-mode.rst

```diff
@@ -39,22 +40,101 @@ The old ``suite.rc`` filename triggers a backward compatibility mode in which:
-you may be able to run it with Cylc 8 without any modifications to begin with.
+you may be able to run it using Cylc 8 without any modifications while in
+compatibility mode.
+
+First, run ``cylc validate`` **with Cylc 7** on your ``suite.rc`` workflow
+to check for deprecation warnings and fix those before validating with Cylc 8.
+See :ref:`below <compat.eg.c7val>` for an example.
+
+.. warning::
+
+   ``cylc validate`` operates on the processed ``suite.rc``, which
+   means it will not detect any deprecated syntax that is inside a
+   currently-unused Jinja2/EmPy ``if...else`` branch.
+
+Some workflows may require modifications to either upgrade to Cylc 8 or make
+interoperable with Cylc 8 backward compatibility mode. Read on for more details.
+
+
+Cylc commands in task scripts
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Check for any use of Cylc commands in task scripting. Some Cylc 7 commands
+have been removed and some others now behave differently.
+However, ``cylc message`` and ``cylc broadcast`` have *not* changed.
+See the :ref:`full list of command line interface changes<MajorChangesCLI>`
+and see :ref:`below <compat.eg.cylc-commands>` for an example.
+
+
+Python 2 to 3
+^^^^^^^^^^^^^
+
+Whereas Cylc 7 runs using Python 2, Cylc 8 runs using Python 3. This affects:
+- modules imported in Jinja2
+- Jinja2 filters, tests and globals
+- custom xtrigger functions
+
+Note that task scripts are not affected - they run in an independent
+environment.
+
+See :ref:`py23` for more information and examples of how to implement
+interoperability if your workflows extend Cylc or Jinja2 with custom Python scripts.
+
+
+Restarting a Cylc 7 workflow
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Cylc 8 cannot *restart* a Cylc 7 workflow mid-run. Instead, :ref:`install
+<Installing-workflows>` the workflow to a new run directory and start it
+from scratch at the right cycle point or task(s):
 
-To see if your workflow is compatible with Cylc 8, run ``cylc validate``
-**using Cylc 7**.
+- ``cylc play --start-cycle-point=<cycle>`` (c.f. Cylc 7 *warm start*), or
+- ``cylc play --start-task=<cycle/task>``   (Cylc 8 can start anywhere in the graph)
 
-If tasks in your workflow call Cylc commands directly it might be necessary to
-modify them to be compatible with command line interface changes
-(unfortunately, this is not possible to detect at validation).
+.. note::
+
+   Any previous-cycle workflow data needed by the new run will need to be
+   manually copied over from the original run directory.
+
+
+Custom remote installation
+^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+If you have used Rose 2019 you may be used to all files and directories inside
+your :term:`source directory` being installed on remote install targets.
+However, by default, Cylc 8 only installs
+:ref:`certain files and directories <RemoteInit>` onto remote install targets.
+See :ref:`below <compat.eg.custom_remote_install>` for an example of how to specify
+custom files to be installed.
```

Comment:
This should go in / be covered by the install section - https://cylc.github.io/cylc-doc/latest/html/7-to-8/major-changes/cylc-install.html#remote-installation